GeneticAlgorithm.py

```diff
@@ -0,0 +1,301 @@
+#Import functions to be used
+import numpy as np #For array manipulations
+import random #for drawing random bits and setting seed
+import sys #Store the biggest number
+
+
+def create_Population(N_pop,n):
+    """
+    Returns a list of solutions that are random and returns N_pop solutions with each having a cost of zero
+        Parameters: 
+            N_pop (int): Population size
+            n (int):  number of facilities or location
+
+        Returns:
+            X_pop (list) - Population of random solutions
+    """
+    cost = 0 
+    X_pop = []
+    for i in range(N_pop):
+        X_i = random.sample(range(n),n) #Sample without replacement assign n facilities to each location
+        X_pop.append([X_i,cost]) #Append the solution and the default cost of zero
+    return X_pop
+
+
+
+
+
+def fitness(distance_data, flow_data, X_pop):
+    """
+    Returns an updated fitness cost for every list in population
+        Parameter:
+            distance_data (matrix): Distance Data matrix
+            flow_data (matrix): Flow Data matrix
+            X_pop (list): N_pop x n matrix storing all N_pop random solutions
+
+        Returns:
+            X_pop (list) -  N_pop x n matrix storing all N_pop random solutions
+    """
+    population = []
+    #Creates a 1xn vector from 0 to n
+    set_location = np.array(range(len(X_pop[0][0]))) #Issue is X_pop[0][0] may not exist
+    for i in range(len(X_pop)): #For every solution in population
+        cost = 0
+        X_i = np.array(X_pop[i][0]) #Store the solution instance for a solution in the population. If the form [1,5,4,2,3]
+
+       
+        for location_i, facility_i in enumerate(X_i):
```

Comment:
you can replace these two for loops that calculate the cost with the function `calculate_objective`. You will have to merge master into this branch first though. 
If you aren't familiar with how to do that, let me know and I can assist

Reply:
I have no clue on how to merge master into the main branch

Reply:
chech out the main branch: `git checkout main` 
then pull the changes from main that are not local: `git pull`
then checkout this branch again: `git checkout Genetic-Algorithm-Function`
then merge the main branch into your branch `git merge main`
then commit and push that as as part of your branch

Reply:
once you do that you can call the function `calculate_objective` instead of this code.